overlord/assertstate/bulk.go

```diff
@@ -242,7 +242,7 @@ func (rpe *resolvePoolError) Error() string {
 	return strings.Join(s, "\n")
 }
 
-func resolvePool(s *state.State, pool *asserts.Pool, checkBeforeCommit func(*asserts.Database, asserts.Backstore) error, userID int, deviceCtx snapstate.DeviceContext) error {
+func resolvePool(s *state.State, pool *asserts.Pool, checkBeforeCommit func(*asserts.Database, asserts.Backstore) error, userID int, deviceCtx snapstate.DeviceContext, auto bool) error {
```

Comment:
While in the other functions (which contain the word "refresh" in their name) it's quite obvious what the "auto" parameter does, here it isn't exactly so. Therefore I'd suggest renaming it to `autoRefresh`.

Reply:
Or `IsAutoRefresh`, I'm not sure which one is better :-)

Reply:
I will comment on this some more later (in the middle of meetings right now) but I am not a big fan ob non-obvious booleans in API. I.e. "foo(..., true)" will not tell much about what the bool means. I would suggest to follow the pattern we use in other function and pass a "resolveOptions struct" (name TBD) in that can be nil (and then the defaults will be used). On that struct we initially only have a single member "IsAutoRefresh" (or something like this). And then it becomes: "foo(..., &resolveOptions{IsAutoRefresh: true\)"